compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs

```diff
@@ -436,6 +438,43 @@ fn quoted_as_type(
     Ok(Value::Type(typ))
 }
 
+// fn as_integer(self) -> Option<(bool, u8)>
+fn type_as_integer(
+    mut arguments: Vec<(Value, Location)>,
+    return_type: Type,
+    location: Location,
+) -> IResult<Value> {
+    check_argument_count(1, &arguments, location)?;
+
+    let value = arguments.pop().unwrap().0;
+    let Value::Type(typ) = value else {
+        panic!("type_as_integer shold have been called with a type");
+    };
+
+    let option_value = if let Type::Integer(sign, bits) = typ {
+        Some(Value::Tuple(vec![
+            Value::Bool(match sign {
+                Signedness::Unsigned => false,
+                Signedness::Signed => true,
+            }),
+            Value::U8(match bits {
+                IntegerBitSize::One => 1,
+                IntegerBitSize::Eight => 8,
+                IntegerBitSize::Sixteen => 16,
+                IntegerBitSize::ThirtyTwo => 32,
+                IntegerBitSize::SixtyFour => 64,
```

Comment:
I think these would fit better as methods on Signedness and IntegerBitSize instead

Reply:
Oh, right! I wasn't sure about doing this and I forgot to add a comment for this. I'll do it 👍

Reply:
Oh, now I remember. I didn't know what to name those methods, or whether to use the `Into` trait for this (I don't know what's the convention in Rust to do this).

Reply:
`is_signed()` and `bit_count()` sound like good names. I like to avoid `Into` conversions generally since things tend to read better without them. I've heard one rule for them could be to include them if you can do a roundtrip on the conversion and not lose any data.